securing_apps/topics/oidc/javascript-adapter.adoc

```diff
@@ -161,6 +161,59 @@ For the Hybrid flow, you need to pass the parameter `flow` with value `hybrid` t
 keycloak.init({ flow: 'hybrid' })
 ----
 
+==== Hybrid Apps with Cordova
+
+Keycloaks support hybird mobile apps developed with https://cordova.apache.org/[Apache Cordova]. The Javascript adapter has two modes for this: `cordova` and `cordova-native`:
+
+The default is `cordova`. The adapter will pick it automatically if no adapter type has been configured and `window.cordova` is present. 
+When logging in, it will open an https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-inappbrowser/[InApp Browser] that lets the user interact with keycloak and afterwards returns to the app by redirecting to http://localhost. 
+Therefore this URL needs to be whitelisted as valid redirect-uri in the client configuration section of the Administration Console.
+
+While this mode is very easy to setup, it also has a few disadvantages:
+The InApp-Browser is a browser embedded in the app and not the phones default browser. 
+Therefore it will have different settings and stored credentials will not be available. It might be slower, especially when rendering more complex themes.
+There are security concerns to consider, before using this mode: 
+It is possible for the app to gain access to the credentials of the user, as it has full control of the browser rendering the login page.
+So do not allow its use in apps you dont trust.
+
+We have an example app that can help you get started: https://github.com/keycloak/keycloak/tree/master/examples/cordova
+
+The alternative mode `cordova-native` takes a different approach.
+It opens the login page in the systems browser. 
+After the user has authenticated, the browser redirects back into the app by a special URL.
+From there, the keycloak adapter can finish the login by reading the code or token from the URL.
+
+You can activate the native mode by passing the adater type `cordova-native` to the `init` method:
+
+[source,javascript]
+----
+keycloak.init({ adapter: 'cordova-native' })
+----
+
+This adapter required two additional plugins:
+
+* https://github.com/google/cordova-plugin-browsertab[cordova-plugin-browsertab]: allows the app to open webpages in the systems browser
+* https://github.com/e-imaxina/cordova-plugin-deeplinks[cordova-plugin-deeplinks]: allow the browser to redirect back to your app by special URLs
+
```

Comment:
_I am making an assumption here, that this applies to both modes of the Javascript adapter. Please correct if I am wrong._

This Javascript adapter requires two additional plugins, regardless of which mode you are using:

* https://github.com/google/cordova-plugin-browsertab[cordova-plugin-browsertab]: allows the app to open webpages in the system's browser
* https://github.com/e-imaxina/cordova-plugin-deeplinks[cordova-plugin-deeplinks]: allows the browser to redirect back to your app using special URLs

Reply:
No, these plugins are only necessary if you're using the native mode.